packages/clay-date-picker/src/index.tsx

```diff
@@ -281,13 +330,62 @@ const ClayDatePicker: React.FunctionComponent<IProps> = React.forwardRef<
 			}
 		};
 
+		/**
+		 * Holds the logic for how to behave when clicking on a date,
+		 * on the DatePicker using date ranges
+		 */
+		const handleDateRangeClicked = (date: Date) => {
+			const [startDate, endDate] = daysSelected;
+
+			if (date < startDate) {
+				setDaysSelected([date, endDate]);
+
+				onValueChange(
+					fromRangeToString(daysSelected, dateFormat),
+					'click'
+				);
+
+				return;
+			}
+
+			setDaysSelected([startDate, date]);
+
+			onValueChange(
+				fromRangeToString([startDate, date], dateFormat),
+				'click'
+			);
+		};
+
 		/**
 		 * Handles the click on element of the day
 		 */
 		const handleDayClicked = (date: Date) => {
-			setDaySelected(date);
+			if (range) {
+				handleDateRangeClicked(date);
+
+				return;
+			}
+
+			setDaysSelected([date, date]);
+
+			const dateFormatted = formatDate(date, dateFormat);
+
+			if (time) {
+				const dateString = formatDate(
+					parseDate(
+						`${dateFormatted} ${currentTime}`,
+						`${dateFormat} ${TIME_FORMAT}`,
+						date
+					),
+					`${dateFormat} ${TIME_FORMAT}`
+				);
 
-			onValueChange(date, 'click');
+				onValueChange(dateString, 'click');
+
+				return;
+			}
+
+			onValueChange(dateFormatted, 'click');
```

Comment:
I still think we can consider the comment I made about it https://github.com/liferay/clay/pull/4012#discussion_r612015851. We can talk more tomorrow because you didn't follow through.